### PR TITLE
[MM-23584][MM-23675] Add first and last name to user profile view & language support

### PR DESCRIPTION
**Pull request by shred86**
#### Summary

Adds the first and last name to the user profile view and reorganizes the order to match the web app and edit profile view. Also removes the username entry since it's already displayed in the second line below the user photo (matches web app). Also, adds language support for display labels.

#### Ticket Link

Fixes: [mattermost-server #14187](https://github.com/mattermost/mattermost-server/issues/14187)
JIRA ticket: [MM-23584](https://mattermost.atlassian.net/browse/MM-23584), [MM-23675](https://mattermost.atlassian.net/browse/MM-23675)

#### Checklist

- [x] Added or updated unit tests (required for all new features)
- [ ] All new/modified APIs include changes to [mattermost-redux](https://github.com/mattermost/mattermost-redux) (please link)
- [x] Has UI changes
- [ ] Includes text changes and localization file updates

#### Device Information
This PR was tested on:
- [iPhone 11, iOS 13.4 (simulator)](https://i.imgur.com/Hto0Lhd.png)
- [Pixel 3, Android 10 (simulator)](https://i.imgur.com/dAj9cpI.png)

### CODE REVIEW

**Comment by shred86** [comment]
> Thanks for the contribution @shred86
> 
> Do you mind posting a screenshot with your changes (if you post both Android and iOS that would be awesome!)
> 
> Finally I just realize that this screen is not using translations for the field names and I filed a ticket for that here https://mattermost.atlassian.net/browse/MM-23675 would you want to work on that too ?

Screenshots as requested:
[iOS 13.4 - iPhone 11](https://i.imgur.com/jeTF0mM.png)
[Android 10 - Pixel 3](https://i.imgur.com/V9ZvqRC.png)

I'm very new to React, React Native and JavaScript but I'd be more than happy to give it a shot. The first and last name fields are just something our organization really needs and it seemed easy enough to add since the properties already existed.

I also noticed there's two options in the system settings for Mattermost that allows a System Admin to enable/disable the ability to see email addresses and full names. I believe the email field checks for this:

```
{this.props.config.ShowEmailAddress === 'true' && this.buildDisplayBlock('email')}
```

However, the first and last name fields do not because there's no prop for `ShowFullName`. It appears this is an issue in the web app as well since it shows the full name regardless of the "show full name setting" set in the system settings. I'm not too familiar with Go either but from what I can tell, [ShowFullNames](https://github.com/mattermost/mattermost-server/blob/4ac0619c90745cba5d053875b2e0f40785040189/api4/user.go#L773) exists but a prop needs to be added to [client.go](https://github.com/mattermost/mattermost-server/blob/1cde88f1479558fbd26dc46e002c7cde43befdb4/config/client.go) so it's accessible from the mobile app.

Once we're able to access `ShowFullName`, I would imagine we would want to check if that setting is `true` before calling `this.buildDisplayBlock()` with `first_name` and `last_name`.

**Comment by shred86** [comment]
Here is a solution I came up with for supporting translations:

```
    buildDisplayBlock = (property) => {
        const {formatMessage} = this.context.intl;
        const {theme, user, isLandscape} = this.props;
        const style = createStyleSheet(theme);
        let label;

        if (user.hasOwnProperty(property) && user[property].length > 0) {
            switch (property) {
                case 'first_name':
                    label = formatMessage({id: 'user.settings.general.firstName', defaultMessage: 'First Name'});
                    break;
                case 'last_name':
                    label = formatMessage({id: 'user.settings.general.lastName', defaultMessage: 'Last Name'});
                    break;
                case 'username':
                    label = formatMessage({id: 'user.settings.general.username', defaultMessage: 'Username'});
                    break;
                case 'nickname':
                    label = formatMessage({id: 'user.settings.general.nickname', defaultMessage: 'Nickname'});
                    break;
                case 'email':
                    label = formatMessage({id: 'user.settings.general.email', defaultMessage: 'Email'});
                    break;
                case 'position':
                    label = formatMessage({id: 'user.settings.general.position', defaultMessage: 'Position'});
            }

            return (
                <View>
                    <Text style={[style.header, padding(isLandscape)]}>{label.toUpperCase()}</Text>
                    <Text style={[style.text, padding(isLandscape)]}>{user[property]}</Text>
                </View>
            );
        }
        
    };
```

The other option is to create separate methods for each field (i.e. `renderFirstName`, `renderLastName`, etc.) which is how edit_profile.js is setup, but it's a lot of repeated code so I thought using a `switch` in this situation made more sense.

I didn't commit this yet since I wasn't sure if this should be a separate PR.

**Comment by enahum** [comment]
Hey @shred86 I think your solution is great!

File: app/screens/user_profile/user_profile.js [hunk 1]
```diff
@@ -351,6 +371,7 @@ const createStyleSheet = makeStyleSheetFromTheme((theme) => {
         header: {
             fontSize: 13,
             fontWeight: '600',
+            textTransform: 'uppercase',
```
Comment (chuttam) [inline app/screens/user_profile/user_profile.js]: 👍

**Comment by shred86** [comment]
Thanks! There is one small outstanding issue that I'd like to fix. Right now, it will always display the first and last name regardless of the "Show Full Name" setting in the system console. I simply need to add one line of code to mattermost-server and mattermost-redux. I'll make those PRs then did you want me to commit the change required to this PR? I just need to update to this:

```
{this.props.config.ShowFullName === 'true' && this.buildDisplayBlock('first_name')}
{this.props.config.ShowFullName === 'true' && this.buildDisplayBlock('last_name')}
```

**Comment by josephbaylon** [comment]
@shred86 @enahum Is this ready to test or should I wait for the other 2 PRs to get merged?

**Comment by enahum** [comment]
@josephbaylon to test this PR at least the server one needs to be merged as this will only work if the server sends down the config to the clients.

**Comment by shred86** [comment]
This PR, as it currently sits, works without any changes to the server PR. Sorry if I caused confusion by referencing the other PRs to this one.

Once the other PRs that add the `ShowFullName` prop on the server side are merged, I’ll make a new PR that adds a check for `this.props.config.ShowFullName` for the first and last name fields.

**Comment by enahum** [comment]
@josephbaylon feel free to test ;)

**Pull request merged**